### PR TITLE
[avmfritz] Added support for HAN-FUN Color and Dimmable bulbs

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added support for HAN-FUN Color and Dimmable bulbs
- Fixed thing type definitions and saturation conversion

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/i18n/avmfritz.properties [hunk 1]
```diff
@@ -39,8 +39,12 @@ thing-type.avmfritz.FRITZ_Powerline_546E_Solo.label = FRITZ!Powerline 546E
 thing-type.avmfritz.FRITZ_Powerline_546E_Solo.description = A FRITZ!Powerline 546E with switchable outlet in stand-alone mode.
 thing-type.avmfritz.HAN_FUN_BLINDS.label = HAN-FUN Blinds
 thing-type.avmfritz.HAN_FUN_BLINDS.description = HAN-FUN blinds (e.g. RolloTron DECT 1213)
+thing-type.avmfritz.HAN_FUN_COLOR_BULB.label = HAN-FUN Color Light
+thing-type.avmfritz.HAN_FUN_COLOR_BULB.description = HAN-FUN color light (e.g SmartHome LED-Lampe E27 (farbig)).
```
Comment (lolodomo) [inline bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/i18n/avmfritz.properties:43]: lamp

File: bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/i18n/avmfritz.properties [hunk 2]
```diff
@@ -39,8 +39,12 @@ thing-type.avmfritz.FRITZ_Powerline_546E_Solo.label = FRITZ!Powerline 546E
...
 thing-type.avmfritz.HAN_FUN_CONTACT.label = HAN-FUN Contact
 thing-type.avmfritz.HAN_FUN_CONTACT.description = HAN-FUN contact (e.g. SmartHome Tür-/Fensterkontakt or SmartHome Bewegungsmelder).
+thing-type.avmfritz.HAN_FUN_DIMMABLE_BULB.label = HAN-FUN Dimmable Light
+thing-type.avmfritz.HAN_FUN_DIMMABLE_BULB.description = HAN-FUN dimmable light (e.g. SmartHome LED-Lampe E27 (warmweiß)).
```
Comment (lolodomo) [inline bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/i18n/avmfritz.properties:47]: lamp

File: bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -314,6 +312,41 @@
 		<config-description-ref uri="thing-type:avmfritz:fritzdevice"/>
 	</thing-type>
 
+	<thing-type id="HAN_FUN_COLOR_BULB">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="fritzbox"/>
+		</supported-bridge-type-refs>
+
+		<label>HAN-FUN Color Light</label>
+		<description>HAN-FUN color light (e.g SmartHome LED-Lampe E27 (farbig)).</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/thing/thing-types.xml:321]: lamp

File: bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -314,6 +312,41 @@
...
+	</thing-type>
+
+
+	<thing-type id="HAN_FUN_DIMMABLE_BULB">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="fritzbox"/>
+		</supported-bridge-type-refs>
+
+		<label>HAN-FUN Dimmable Light</label>
+		<description>HAN-FUN dimmable light (e.g. SmartHome LED-Lampe E27 (warmweiß)).</description>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/thing/thing-types.xml:339]: lamp

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.avmfritz/README.md [hunk 5]
```diff
@@ -92,6 +92,8 @@ The following sensors have been successfully tested using FRITZ!OS 7 for FRITZ!B
 - [SmartHome Zwischenstecker außen](https://www.smarthome.de/geraete/smarthome-zwischenstecker-aussen-schwarz) - a switchable outdoor outlet (thing type `HAN_FUN_ON_OFF`)
 - [Rollotron DECT 1213](https://www.rademacher.de/shop/rollladen-sonnenschutz/elektrischer-gurtwickler/rollotron-dect-1213) - an electronic belt winder (thing type `HAN_FUN_BLINDS`)
 - [Becker BoxCTRL](https://becker-antriebe.shop/) - a radio controlled roller shutter drive (thing type `HAN_FUN_BLINDS`)
+- SmartHome LED-Lampe E27 (farbig) - a dimmable colorized light bulb (thing type `HAN_FUN_COLOR_BULB`)
+- SmartHome LED-Lampe E27 (warmweiß) - a dimmable light bulb (thing type `HAN_FUN_DIMMABLE_BULB`)
```
[the remark this answers is not in the document]
Comment (cweitkamp) [reply bundles/org.openhab.binding.avmfritz/README.md:96]: Thanks for pointing this out. If you check other description of HAN-FUN thing types, translations, etc. you will find more German product descriptions. This is by intention as it is the original product name of the device.

https://github.com/openhab/openhab-addons/blob/78cc8fcaa3bb584345ae2a65141df66bf5982d48/bundles/org.openhab.binding.avmfritz/src/main/resources/OH-INF/thing/thing-types.xml#L264

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.avmfritz/README.md:96]: Ok

**Review by lolodomo — APPROVED** [review]

**Pull request merged**